core/codegen/src/ident.rs

```diff
@@ -0,0 +1,99 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[repr(transparent)]
+#[derive(Clone, Debug, Eq, Ord)]
+pub struct Ident(Ident2);
+
+impl Ident {
+    pub fn is_raw(&self) -> bool {
+        self.0.to_string().starts_with("r#")
+    }
+
+    pub fn name(&self) -> String {
+        self.0.unraw().to_string()
+    }
+
+    pub fn prepend(&self, string: &str) -> Ident {
+        // TODO: want to use r#{}{}, but that fails (`"r#__rocket_param_enum"` is not a valid identifier)
+        Ident::from(syn::Ident::new(&format!("{}{}", string, self.unraw()), self.span()))
+    }
+
+    pub fn span(&self) -> Span {
+        self.0.span()
+    }
+}
+
+impl std::ops::Deref for Ident {
+    type Target = Ident2;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for Ident {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl From<Ident2> for Ident {
+    fn from(ident: Ident2) -> Ident {
+        Ident(ident)
+    }
+}
+
+impl From<Ident> for Ident2 {
+    fn from(ident: Ident) -> Ident2 {
+        ident.0
+    }
+}
+
+impl std::hash::Hash for Ident {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.name().hash(state)
+    }
+}
+
+impl PartialEq for Ident {
+    fn eq(&self, other: &Ident) -> bool {
+        self.name() == other.name()
+    }
+}
+
+impl<T: ?Sized + AsRef<str>> PartialEq<T> for Ident {
+    fn eq(&self, other: &T) -> bool {
+        self.name() == other.as_ref()
+    }
+}
+
+impl PartialOrd for Ident {
```

Comment:
Do we use this?

Reply:
No; I kept it for feature parity with `Ident` from `proc_macro2`.

Reply:
Is this the same as `#[derive(PartialOrd)]`?

Reply:
And, should we be comparing `name()` instead, since that's what `PartialEq` is defined on?

Reply:
I've actually removed this - I expect sorting/comparing `Ident`s to be very unlikely and should be handled more thoughtfully.